rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template

```diff
@@ -182,7 +182,7 @@ convert_ros_to_dds(const void * untyped_ros_message, void * untyped_dds_message)
       return false;
     }
     DDS_Long length = static_cast<DDS_Long>(size);
-    dds_message->@(field.name)_.length(length);
+    dds_message->@(field.name)_.ensure_length(length, 2 * length);
```

Comment:
I am not sure if the `dds_message` instance exists longer then for a single serialization but if not I think `(length, length)` would be better to not allocate more memory then needed.